scripts/example_script.py

```diff
@@ -0,0 +1,241 @@
+        except HttpError as err:
+            print_http_error(err)
+            return
+        folder_id_list = get_field(folders, '_id')
+
+        if resource_type is 'item':
+            try:
+                data = client.listItem(parent_folder)
+
+                resource_list.extend(data)
+            except HttpError as err:
+                print_http_error(err)
+                exit()
+        elif resource_type is 'folder':
+            resource_list.extend(folders)
+        else:
+            return
+    for folder_id in folder_id_list:
+        recurse_get_resource(client, folder_id, resource_type, resource_list)
+
+
+def print_field(data, str_key):
+    for i in range(len(data)):
+        print(data[i][str_key])
+
+
+def get_field(data, str_key):
+    return [i[str_key] for i in data]
+
+
+def turn_into_a_large_image(girder_client, file_id):
+    # first check if the file is already a large image
+    item_data = None
+    try:
+        item_data = gc.getItem(file_id)
+    except HttpError as err:
+        print_http_error(err)
+        print('bad item id ')
+    # if the item is already a large image return
+
+    if 'largeImage' in item_data:
+        return
+    try:
+        girder_client.post('item/%s/tiles' % file_id)
+    except HttpError as err:
+        print_http_error(err)
+
+
+def isEmpty(girder_client, folder):
+    item_list = []
+    recurse_get_resource(girder_client, folder, 'item', item_list)
+    if len(item_list) == 0:
+        return True
+    else:
+        return False
+girder_location = 'http://localhost:8080/api/v1'
+
+collection_name = 'DG TCGA'
+collection_id = None
+request_url = 'resource/lookup?path=collection/%s' % (collection_name)
+collection_data = None
+
+new_folder_name = 'GBM'
+new_folder_desc = 'orginizing patient data'
+new_folder_id = None
+
+img_type = '.jpeg'
+
+# login to girder, you will be prompted for credentials
+gc = girder_client.GirderClient(apiUrl=girder_location)
+gc.authenticate(interactive=True)
+
+# get the id of the desired collection
+try:
+    collection_data = gc.getResource(request_url)
+except HttpError as err:
+    print_http_error(err)
+    exit()
+
+collection_id = collection_data['_id']
+new_folder_data = None
+# create a folder under this collection or load a preexisting folder that exists
+# with the same name and parent
+try:
+    new_folder_data = gc.load_or_create_folder(new_folder_name,
+                                               collection_id, 'collection')
+except HttpError as err:
+    print_http_error(err)
+    exit()
+new_folder_id = new_folder_data['_id']
+
+# get all folders in the collection
+collection_composition = None
+try:
+    collection_composition = gc.listFolder(collection_id, 'collection')
+except HttpError as err:
+    print_http_error(err)
+    print ('exited')
+    exit()
+
+# get the ids of all folders under the collection
+collection_folders = get_field(collection_composition, '_id')
+item_list = []
```

Comment:
Rather than have `recurse_get_resource` modify the `item_list` in-place, it might be better to just return a new list. Even if it's slightly less efficient when you want to extend an existing list, it's a lot more clear to understand.